pkg/platform/catalog_fetcher.go

```diff
@@ -0,0 +1,11 @@
+package platform
+
+// CatalogFetcher provides a way to add a hook for platform specific way of refetching the service broker catalog on each
+// run of the registration task. If the platform that this proxy represents already handles that, you don't
+// have to implement this interface
+//go:generate counterfeiter . CatalogFetcher
+type CatalogFetcher interface {
```

Comment:
Is this extra interface needed as a separate entity? Can the method (not called Fetch but Sync/Refresh) be in the Client?

Reply:
I think seperate would be better as some proxy implementations might not need it - its platform specific how a platform does catalog refetching - and in some platforms (I suppose K8S) - you wouldnt need programatic logic to trigger refetching. If a proxy does not need to implement "Fetch" and fetch is part of Client interface, we would force the implementation to implement empty method